lib/mondo/transaction.rb

```diff
@@ -28,6 +27,15 @@ def save_metadata
       self.client.api_patch("/transactions/#{self.id}", metadata: self.metadata)
     end
 
+    def merchant(opts={})
+      if raw_data['merchant'].kind_of?(Hash)
+        ::Mondo::Merchant.new(raw_data['merchant'], client)
+      else
+        self.raw_data['merchant'] = self.client.transaction(self.id, expand: [:merchant]).raw_data['merchant']
+        merchant(opts)
```

Comment:
If we ever return anything other than a hash for `merchant`, this will explode :boom: I think this may sometimes happen even during normal operation

Would probably just copy line 32 instead of a recursive call.